### ISSUE TITLE
Unexpected errors are reported when pattern locals conflict with regular locals declared within the same statement

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#9612]
The following test fails 
```
        [Fact]
        public void LocalDeclarationStmt_01()
        {
            var source =
@"
public class X
{
    public static void Main()
    {
        object x1 = Dummy(null,
                          new object() is var x1, x1);
    }

    static object Dummy(object x, params object[] y) {return x;}
}
";
            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
            compilation.VerifyDiagnostics(
// (7,47): error CS0136: A local or parameter named 'x1' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter
//                           new object() is var x1, x1);
Diagnostic(ErrorCode.ERR_LocalIllegallyOverrides, "x1").WithArguments("x1").WithLocation(7, 47)
                );
        }
```

Unexpected error:
```
// (6,16): error CS0136: A local or parameter named 'x1' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter
//         object x1 = Dummy(null,
Diagnostic(ErrorCode.ERR_LocalIllegallyOverrides, "x1").WithArguments("x1").WithLocation(6, 16),
```

The following test fails:
```
        [Fact]
        public void LocalDeclarationStmt_02()
        {
            var source =
@"
public class X
{
    public static void Main()
    {
        object l1 = Dummy(new object() is var x1),
               x1;

        x1 = null;
        System.Console.WriteLine(x1);
    }

    static object Dummy(object x, params object[] y) {return x;}
}
";
            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
            compilation.VerifyDiagnostics(
// (6,47): error CS0136: A local or parameter named 'x1' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter
//         object l1 = Dummy(new object() is var x1),
Diagnostic(ErrorCode.ERR_LocalIllegallyOverrides, "x1").WithArguments("x1").WithLocation(6, 47)
                );
```

Unexpected error:
```
// (7,16): error CS0136: A local or parameter named 'x1' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter
//                x1;
Diagnostic(ErrorCode.ERR_LocalIllegallyOverrides, "x1").WithArguments("x1").WithLocation(7, 16)
```

Related to #8817.
